eng/test_tools.txt

```diff
@@ -22,4 +22,8 @@ pyyaml==6.0.1
 urllib3==1.26.16
 PyGithub==1.59.1
 ConfigArgParse==1.2.3
-six==1.14.0
+six==1.16.0
+
+# 312 compatibility
+https://docsupport.blob.core.windows.net/repackaged/cffi-1.15.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl; python_version >= '3.12'
+https://docsupport.blob.core.windows.net/repackaged/aiohttp-4.0.0a2.dev0-cp312-cp312-linux_x86_64.whl; python_version >= '3.12'
```

Comment:
aiohttp 3.9.0b2 has fewer breaking changes and is the newer version for 3.12 support
https://pypi.org/project/aiohttp/#history See https://github.com/aio-libs/aiohttp/issues/7675

Reply:
Thanks for the heads up! Unfortunately the new version broke a few tests. 😂 Removed pin in favor of allowing pip resolution from pypi.

Reply:
Hmm. Unfortunately I can't seem to properly build the `3.8.6` wheels on python 3.12. Looks like a build failure :( 

Going to have to revert back to these while `aiohttp` figures out what's up.